// ng2-bootstrap/modal/modal.component.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
import * as import0 from 'ng2-bootstrap/modal/modal.component';
import * as import1 from '@angular/core/src/change_detection/change_detection_util';
import * as import3 from '@angular/core/src/linker/view_utils';
var Wrapper_ModalDirective = (function () {
    function Wrapper_ModalDirective(p0, p1, p2, p3) {
        this._changed = false;
        this.context = new import0.ModalDirective(p0, p1, p2, p3);
        this._expr_0 = import1.UNINITIALIZED;
    }
    Wrapper_ModalDirective.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_ModalDirective.prototype.ngOnDestroy = function () {
        this.context.ngOnDestroy();
        (this.subscription0 && this.subscription0.unsubscribe());
        (this.subscription1 && this.subscription1.unsubscribe());
        (this.subscription2 && this.subscription2.unsubscribe());
        (this.subscription3 && this.subscription3.unsubscribe());
    };
    Wrapper_ModalDirective.prototype.check_config = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this._changed = true;
            this.context.config = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_ModalDirective.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_ModalDirective.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_ModalDirective.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        if ((eventName == 'click')) {
            var pd_sub_0 = (this.context.onClick($event) !== false);
            result = (pd_sub_0 && result);
        }
        if ((eventName == 'keydown.esc')) {
            var pd_sub_1 = (this.context.onEsc() !== false);
            result = (pd_sub_1 && result);
        }
        return result;
    };
    Wrapper_ModalDirective.prototype.subscribe = function (view, _eventHandler, emit0, emit1, emit2, emit3) {
        this._eventHandler = _eventHandler;
        if (emit0) {
            (this.subscription0 = this.context.onShow.subscribe(_eventHandler.bind(view, 'onShow')));
        }
        if (emit1) {
            (this.subscription1 = this.context.onShown.subscribe(_eventHandler.bind(view, 'onShown')));
        }
        if (emit2) {
            (this.subscription2 = this.context.onHide.subscribe(_eventHandler.bind(view, 'onHide')));
        }
        if (emit3) {
            (this.subscription3 = this.context.onHidden.subscribe(_eventHandler.bind(view, 'onHidden')));
        }
    };
    return Wrapper_ModalDirective;
}());
export { Wrapper_ModalDirective };
