// ng2-bootstrap/modal/modal-backdrop.component.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import * as import0 from 'ng2-bootstrap/modal/modal-backdrop.component';
import * as import1 from '@angular/core/src/linker/view';
import * as import3 from '@angular/core/src/linker/view_utils';
import * as import4 from '@angular/core/src/metadata/view';
import * as import5 from '@angular/core/src/linker/view_type';
import * as import6 from '@angular/core/src/change_detection/constants';
import * as import7 from '@angular/core/src/linker/component_factory';
import * as import8 from '@angular/core/src/linker/element_ref';
var Wrapper_ModalBackdropComponent = (function () {
    function Wrapper_ModalBackdropComponent(p0, p1) {
        this._changed = false;
        this.context = new import0.ModalBackdropComponent(p0, p1);
    }
    Wrapper_ModalBackdropComponent.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_ModalBackdropComponent.prototype.ngOnDestroy = function () {
    };
    Wrapper_ModalBackdropComponent.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_ModalBackdropComponent.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_ModalBackdropComponent.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        return result;
    };
    Wrapper_ModalBackdropComponent.prototype.subscribe = function (view, _eventHandler) {
        this._eventHandler = _eventHandler;
    };
    return Wrapper_ModalBackdropComponent;
}());
export { Wrapper_ModalBackdropComponent };
var renderType_ModalBackdropComponent_Host = import3.createRenderComponentType('', 0, import4.ViewEncapsulation.None, [], {});
var View_ModalBackdropComponent_Host0 = (function (_super) {
    __extends(View_ModalBackdropComponent_Host0, _super);
    function View_ModalBackdropComponent_Host0(viewUtils, parentView, parentIndex, parentElement) {
        return _super.call(this, View_ModalBackdropComponent_Host0, renderType_ModalBackdropComponent_Host, import5.ViewType.HOST, viewUtils, parentView, parentIndex, parentElement, import6.ChangeDetectorStatus.CheckAlways) || this;
    }
    View_ModalBackdropComponent_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer, 'bs-modal-backdrop', new import3.InlineArray2(2, 'class', 'modal-backdrop'), rootSelector, null);
        this.compView_0 = new View_ModalBackdropComponent0(this.viewUtils, this, 0, this._el_0);
        this._ModalBackdropComponent_0_3 = new Wrapper_ModalBackdropComponent(new import8.ElementRef(this._el_0), this.renderer);
        this.compView_0.create(this._ModalBackdropComponent_0_3.context);
        this.init(this._el_0, (this.renderer.directRenderer ? null : [this._el_0]), null);
        return new import7.ComponentRef_(0, this, this._el_0, this._ModalBackdropComponent_0_3.context);
    };
    View_ModalBackdropComponent_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import0.ModalBackdropComponent) && (0 === requestNodeIndex))) {
            return this._ModalBackdropComponent_0_3.context;
        }
        return notFoundResult;
    };
    View_ModalBackdropComponent_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        this._ModalBackdropComponent_0_3.ngDoCheck(this, this._el_0, throwOnChange);
        this.compView_0.internalDetectChanges(throwOnChange);
    };
    View_ModalBackdropComponent_Host0.prototype.destroyInternal = function () {
        this.compView_0.destroy();
    };
    View_ModalBackdropComponent_Host0.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    return View_ModalBackdropComponent_Host0;
}(import1.AppView));
export var ModalBackdropComponentNgFactory = new import7.ComponentFactory('bs-modal-backdrop', View_ModalBackdropComponent_Host0, import0.ModalBackdropComponent);
var styles_ModalBackdropComponent = [];
var renderType_ModalBackdropComponent = import3.createRenderComponentType('', 0, import4.ViewEncapsulation.None, styles_ModalBackdropComponent, {});
var View_ModalBackdropComponent0 = (function (_super) {
    __extends(View_ModalBackdropComponent0, _super);
    function View_ModalBackdropComponent0(viewUtils, parentView, parentIndex, parentElement) {
        return _super.call(this, View_ModalBackdropComponent0, renderType_ModalBackdropComponent, import5.ViewType.COMPONENT, viewUtils, parentView, parentIndex, parentElement, import6.ChangeDetectorStatus.CheckAlways) || this;
    }
    View_ModalBackdropComponent0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.parentElement);
        this.init(null, (this.renderer.directRenderer ? null : []), null);
        return null;
    };
    return View_ModalBackdropComponent0;
}(import1.AppView));
export { View_ModalBackdropComponent0 };
