// ng2-file-upload/file-upload/file-select.directive.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
import * as import0 from 'ng2-file-upload/file-upload/file-select.directive';
import * as import1 from '@angular/core/src/change_detection/change_detection_util';
import * as import3 from '@angular/core/src/linker/view_utils';
var Wrapper_FileSelectDirective = (function () {
    function Wrapper_FileSelectDirective(p0) {
        this._changed = false;
        this.context = new import0.FileSelectDirective(p0);
        this._expr_0 = import1.UNINITIALIZED;
    }
    Wrapper_FileSelectDirective.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_FileSelectDirective.prototype.ngOnDestroy = function () {
    };
    Wrapper_FileSelectDirective.prototype.check_uploader = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this._changed = true;
            this.context.uploader = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_FileSelectDirective.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_FileSelectDirective.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_FileSelectDirective.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        if ((eventName == 'change')) {
            var pd_sub_0 = (this.context.onChange() !== false);
            result = (pd_sub_0 && result);
        }
        return result;
    };
    Wrapper_FileSelectDirective.prototype.subscribe = function (view, _eventHandler) {
        this._eventHandler = _eventHandler;
    };
    return Wrapper_FileSelectDirective;
}());
export { Wrapper_FileSelectDirective };
