// ng2-bootstrap/timepicker/timepicker.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from 'ng2-bootstrap/timepicker/timepicker.component';
import * as import1 from '@angular/core/src/change_detection/change_detection_util';
import * as import2 from '@angular/core/src/linker/view';
import * as import3 from '@angular/core/src/linker/view_utils';
import * as import4 from '@angular/core/src/render/api';
import * as import5 from '@angular/core/src/metadata/view';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/constants';
import * as import8 from '@angular/core/src/linker/component_factory';
import * as import9 from 'ng2-bootstrap/timepicker/timepicker.config';
import * as import10 from '@angular/forms/src/directives/control_value_accessor';
import * as import11 from '../../@angular/common/src/directives/ng_class.ngfactory';
import * as import12 from '@angular/core/src/linker/view_container';
import * as import13 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import14 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import15 from '@angular/core/src/linker/element_ref';
import * as import16 from '@angular/common/src/directives/ng_class';
import * as import17 from '../../@angular/common/src/directives/ng_if.ngfactory';
import * as import18 from '../../@angular/forms/src/directives/default_value_accessor.ngfactory';
import * as import19 from '../../@angular/forms/src/directives/validators.ngfactory';
import * as import20 from '../../@angular/forms/src/directives/ng_model.ngfactory';
import * as import21 from '../../@angular/forms/src/directives/ng_control_status.ngfactory';
import * as import22 from '@angular/core/src/linker/template_ref';
import * as import23 from '@angular/common/src/directives/ng_if';
import * as import24 from '@angular/forms/src/directives/default_value_accessor';
import * as import25 from '@angular/forms/src/directives/validators';
import * as import26 from '@angular/forms/src/validators';
import * as import27 from '@angular/forms/src/directives/ng_model';
import * as import28 from '@angular/forms/src/directives/ng_control';
import * as import29 from '@angular/forms/src/directives/ng_control_status';
export class Wrapper_TimepickerComponent {
  /*private*/ _eventHandler:Function;
  context:import0.TimepickerComponent;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  /*private*/ _expr_6:any;
  /*private*/ _expr_7:any;
  /*private*/ _expr_8:any;
  /*private*/ _expr_9:any;
  constructor(p0:any) {
    this._changed = false;
    this.context = new import0.TimepickerComponent(p0);
    this._expr_0 = import1.UNINITIALIZED;
    this._expr_1 = import1.UNINITIALIZED;
    this._expr_2 = import1.UNINITIALIZED;
    this._expr_3 = import1.UNINITIALIZED;
    this._expr_4 = import1.UNINITIALIZED;
    this._expr_5 = import1.UNINITIALIZED;
    this._expr_6 = import1.UNINITIALIZED;
    this._expr_7 = import1.UNINITIALIZED;
    this._expr_8 = import1.UNINITIALIZED;
    this._expr_9 = import1.UNINITIALIZED;
  }
  ngOnDetach(view:import2.AppView<any>,componentView:import2.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
  }
  check_hourStep(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.hourStep = currValue;
      this._expr_0 = currValue;
    }
  }
  check_minuteStep(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.minuteStep = currValue;
      this._expr_1 = currValue;
    }
  }
  check_readonlyInput(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.readonlyInput = currValue;
      this._expr_2 = currValue;
    }
  }
  check_mousewheel(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.mousewheel = currValue;
      this._expr_3 = currValue;
    }
  }
  check_arrowkeys(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.arrowkeys = currValue;
      this._expr_4 = currValue;
    }
  }
  check_showSpinners(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_5,currValue))) {
      this._changed = true;
      this.context.showSpinners = currValue;
      this._expr_5 = currValue;
    }
  }
  check_min(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_6,currValue))) {
      this._changed = true;
      this.context.min = currValue;
      this._expr_6 = currValue;
    }
  }
  check_max(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_7,currValue))) {
      this._changed = true;
      this.context.max = currValue;
      this._expr_7 = currValue;
    }
  }
  check_meridians(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_8,currValue))) {
      this._changed = true;
      this.context.meridians = currValue;
      this._expr_8 = currValue;
    }
  }
  check_showMeridian(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_9,currValue))) {
      this._changed = true;
      this.context.showMeridian = currValue;
      this._expr_9 = currValue;
    }
  }
  ngDoCheck(view:import2.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    if (!throwOnChange) { if ((view.numberOfChecks === 0)) { this.context.ngOnInit(); } }
    return changed;
  }
  checkHost(view:import2.AppView<any>,componentView:import2.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import2.AppView<any>,_eventHandler:any):void {
    this._eventHandler = _eventHandler;
  }
}
var renderType_TimepickerComponent_Host:import4.RenderComponentType = import3.createRenderComponentType('',0,import5.ViewEncapsulation.None,([] as any[]),{});
class View_TimepickerComponent_Host0 extends import2.AppView<any> {
  _el_0:any;
  compView_0:import2.AppView<import0.TimepickerComponent>;
  _TimepickerComponent_0_3:Wrapper_TimepickerComponent;
  __NG_VALUE_ACCESSOR_0_4:any[];
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_TimepickerComponent_Host0,renderType_TimepickerComponent_Host,import6.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways);
  }
  get _NG_VALUE_ACCESSOR_0_4():any[] {
    if ((this.__NG_VALUE_ACCESSOR_0_4 == null)) { (this.__NG_VALUE_ACCESSOR_0_4 = [this._TimepickerComponent_0_3.context]); }
    return this.__NG_VALUE_ACCESSOR_0_4;
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer,'timepicker',import3.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_TimepickerComponent0(this.viewUtils,this,0,this._el_0);
    this._TimepickerComponent_0_3 = new Wrapper_TimepickerComponent(this.injectorGet(import9.TimepickerConfig,this.parentIndex));
    this.compView_0.create(this._TimepickerComponent_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import8.ComponentRef_<any>(0,this,this._el_0,this._TimepickerComponent_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import0.TimepickerComponent) && (0 === requestNodeIndex))) { return this._TimepickerComponent_0_3.context; }
    if (((token === import10.NG_VALUE_ACCESSOR) && (0 === requestNodeIndex))) { return this._NG_VALUE_ACCESSOR_0_4; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._TimepickerComponent_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const TimepickerComponentNgFactory:import8.ComponentFactory<import0.TimepickerComponent> = new import8.ComponentFactory<import0.TimepickerComponent>('timepicker',View_TimepickerComponent_Host0,import0.TimepickerComponent);
const styles_TimepickerComponent:any[] = ([] as any[]);
class View_TimepickerComponent1 extends import2.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import11.Wrapper_NgClass;
  _map_2:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import12.ViewContainer) {
    super(View_TimepickerComponent1,renderType_TimepickerComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_2 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {hidden: p0};
    });
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'td',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_0_3 = new import11.Wrapper_NgClass(this.parentView.parentView.injectorGet(import13.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import14.KeyValueDiffers,this.parentView.parentIndex),new import15.ElementRef(this._el_0),this.renderer);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import16.NgClass) && (0 === requestNodeIndex))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this._map_2(!this.parentView.context.showMeridian);
    this._NgClass_0_3.check_ngClass(currVal_0_0_0,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_TimepickerComponent2 extends import2.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import11.Wrapper_NgClass;
  _el_1:any;
  _NgClass_1_3:import11.Wrapper_NgClass;
  _text_2:any;
  _map_5:any;
  _map_6:any;
  /*private*/ _expr_7:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import12.ViewContainer) {
    super(View_TimepickerComponent2,renderType_TimepickerComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_5 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {hidden: p0};
    });
    this._map_6 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {disabled: p0};
    });
    this._expr_7 = import1.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'td',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_0_3 = new import11.Wrapper_NgClass(this.parentView.parentView.injectorGet(import13.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import14.KeyValueDiffers,this.parentView.parentIndex),new import15.ElementRef(this._el_0),this.renderer);
    this._el_1 = import3.createRenderElement(this.renderer,this._el_0,'button',new import3.InlineArray4(4,'class','btn btn-default text-center','type','button'),(null as any));
    this._NgClass_1_3 = new import11.Wrapper_NgClass(this.parentView.parentView.injectorGet(import13.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import14.KeyValueDiffers,this.parentView.parentIndex),new import15.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'',(null as any));
    var disposable_0:Function = import3.subscribeToRenderElement(this,this._el_1,new import3.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_1));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._el_1,
      this._text_2
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import16.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 2)))) { return this._NgClass_1_3.context; }
    if (((token === import16.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 2)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this._map_5(!this.parentView.context.showMeridian);
    this._NgClass_0_3.check_ngClass(currVal_0_0_0,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_1_0_0:any = 'btn btn-default text-center';
    this._NgClass_1_3.check_klass(currVal_1_0_0,throwOnChange,false);
    const currVal_1_0_1:any = this._map_6((this.parentView.context.noToggleMeridian() || this.parentView.context.readonlyInput));
    this._NgClass_1_3.check_ngClass(currVal_1_0_1,throwOnChange,false);
    this._NgClass_1_3.ngDoCheck(this,this._el_1,throwOnChange);
    const currVal_7:any = import3.inlineInterpolate(1,'',this.parentView.context.meridian,'');
    if (import3.checkBinding(throwOnChange,this._expr_7,currVal_7)) {
      this.renderer.setText(this._text_2,currVal_7);
      this._expr_7 = currVal_7;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_1(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.toggleMeridian()) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_TimepickerComponent3 extends import2.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import11.Wrapper_NgClass;
  _map_2:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import12.ViewContainer) {
    super(View_TimepickerComponent3,renderType_TimepickerComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_2 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {hidden: p0};
    });
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'td',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_0_3 = new import11.Wrapper_NgClass(this.parentView.parentView.injectorGet(import13.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import14.KeyValueDiffers,this.parentView.parentIndex),new import15.ElementRef(this._el_0),this.renderer);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import16.NgClass) && (0 === requestNodeIndex))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this._map_2(!this.parentView.context.showMeridian);
    this._NgClass_0_3.check_ngClass(currVal_0_0_0,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
var renderType_TimepickerComponent:import4.RenderComponentType = import3.createRenderComponentType('',0,import5.ViewEncapsulation.None,styles_TimepickerComponent,{});
export class View_TimepickerComponent0 extends import2.AppView<import0.TimepickerComponent> {
  _text_0:any;
  _el_1:any;
  _text_2:any;
  _el_3:any;
  _text_4:any;
  _el_5:any;
  _NgClass_5_3:import11.Wrapper_NgClass;
  _text_6:any;
  _el_7:any;
  _el_8:any;
  _NgClass_8_3:import11.Wrapper_NgClass;
  _el_9:any;
  _text_10:any;
  _el_11:any;
  _text_12:any;
  _text_13:any;
  _el_14:any;
  _el_15:any;
  _NgClass_15_3:import11.Wrapper_NgClass;
  _el_16:any;
  _text_17:any;
  _anchor_18:any;
  /*private*/ _vc_18:import12.ViewContainer;
  _TemplateRef_18_5:any;
  _NgIf_18_6:import17.Wrapper_NgIf;
  _text_19:any;
  _text_20:any;
  _el_21:any;
  _text_22:any;
  _el_23:any;
  _NgClass_23_3:import11.Wrapper_NgClass;
  _text_24:any;
  _el_25:any;
  _DefaultValueAccessor_25_3:import18.Wrapper_DefaultValueAccessor;
  _MaxLengthValidator_25_4:import19.Wrapper_MaxLengthValidator;
  _NG_VALIDATORS_25_5:any[];
  _NG_VALUE_ACCESSOR_25_6:any[];
  _NgModel_25_7:import20.Wrapper_NgModel;
  _NgControl_25_8:any;
  _NgControlStatus_25_9:import21.Wrapper_NgControlStatus;
  _text_26:any;
  _text_27:any;
  _el_28:any;
  _text_29:any;
  _text_30:any;
  _el_31:any;
  _NgClass_31_3:import11.Wrapper_NgClass;
  _text_32:any;
  _el_33:any;
  _DefaultValueAccessor_33_3:import18.Wrapper_DefaultValueAccessor;
  _MaxLengthValidator_33_4:import19.Wrapper_MaxLengthValidator;
  _NG_VALIDATORS_33_5:any[];
  _NG_VALUE_ACCESSOR_33_6:any[];
  _NgModel_33_7:import20.Wrapper_NgModel;
  _NgControl_33_8:any;
  _NgControlStatus_33_9:import21.Wrapper_NgControlStatus;
  _text_34:any;
  _text_35:any;
  _anchor_36:any;
  /*private*/ _vc_36:import12.ViewContainer;
  _TemplateRef_36_5:any;
  _NgIf_36_6:import17.Wrapper_NgIf;
  _text_37:any;
  _text_38:any;
  _el_39:any;
  _NgClass_39_3:import11.Wrapper_NgClass;
  _text_40:any;
  _el_41:any;
  _el_42:any;
  _NgClass_42_3:import11.Wrapper_NgClass;
  _el_43:any;
  _text_44:any;
  _el_45:any;
  _text_46:any;
  _text_47:any;
  _el_48:any;
  _el_49:any;
  _NgClass_49_3:import11.Wrapper_NgClass;
  _el_50:any;
  _text_51:any;
  _anchor_52:any;
  /*private*/ _vc_52:import12.ViewContainer;
  _TemplateRef_52_5:any;
  _NgIf_52_6:import17.Wrapper_NgIf;
  _text_53:any;
  _text_54:any;
  _text_55:any;
  _text_56:any;
  _map_88:any;
  _map_89:any;
  _map_90:any;
  _map_91:any;
  /*private*/ _expr_92:any;
  _map_93:any;
  /*private*/ _expr_94:any;
  _map_95:any;
  _map_96:any;
  _map_97:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_TimepickerComponent0,renderType_TimepickerComponent,import6.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import7.ChangeDetectorStatus.CheckAlways);
    this._map_88 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {hidden: p0};
    });
    this._map_89 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {disabled: p0};
    });
    this._map_90 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {disabled: p0};
    });
    this._map_91 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {'has-error': p0};
    });
    this._expr_92 = import1.UNINITIALIZED;
    this._map_93 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {'has-error': p0};
    });
    this._expr_94 = import1.UNINITIALIZED;
    this._map_95 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {hidden: p0};
    });
    this._map_96 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {disabled: p0};
    });
    this._map_97 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {disabled: p0};
    });
  }
  createInternal(rootSelector:string):import8.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    this._el_1 = import3.createRenderElement(this.renderer,parentRenderNode,'table',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_2 = this.renderer.createText(this._el_1,'\n      ',(null as any));
    this._el_3 = import3.createRenderElement(this.renderer,this._el_1,'tbody',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_4 = this.renderer.createText(this._el_3,'\n        ',(null as any));
    this._el_5 = import3.createRenderElement(this.renderer,this._el_3,'tr',new import3.InlineArray2(2,'class','text-center'),(null as any));
    this._NgClass_5_3 = new import11.Wrapper_NgClass(this.parentView.injectorGet(import13.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import14.KeyValueDiffers,this.parentIndex),new import15.ElementRef(this._el_5),this.renderer);
    this._text_6 = this.renderer.createText(this._el_5,'\n          ',(null as any));
    this._el_7 = import3.createRenderElement(this.renderer,this._el_5,'td',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._el_8 = import3.createRenderElement(this.renderer,this._el_7,'a',new import3.InlineArray2(2,'class','btn btn-link'),(null as any));
    this._NgClass_8_3 = new import11.Wrapper_NgClass(this.parentView.injectorGet(import13.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import14.KeyValueDiffers,this.parentIndex),new import15.ElementRef(this._el_8),this.renderer);
    this._el_9 = import3.createRenderElement(this.renderer,this._el_8,'span',new import3.InlineArray2(2,'class','glyphicon glyphicon-chevron-up'),(null as any));
    this._text_10 = this.renderer.createText(this._el_5,'\n          ',(null as any));
    this._el_11 = import3.createRenderElement(this.renderer,this._el_5,'td',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_12 = this.renderer.createText(this._el_11,' ',(null as any));
    this._text_13 = this.renderer.createText(this._el_5,'\n          ',(null as any));
    this._el_14 = import3.createRenderElement(this.renderer,this._el_5,'td',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._el_15 = import3.createRenderElement(this.renderer,this._el_14,'a',new import3.InlineArray2(2,'class','btn btn-link'),(null as any));
    this._NgClass_15_3 = new import11.Wrapper_NgClass(this.parentView.injectorGet(import13.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import14.KeyValueDiffers,this.parentIndex),new import15.ElementRef(this._el_15),this.renderer);
    this._el_16 = import3.createRenderElement(this.renderer,this._el_15,'span',new import3.InlineArray2(2,'class','glyphicon glyphicon-chevron-up'),(null as any));
    this._text_17 = this.renderer.createText(this._el_5,'\n          ',(null as any));
    this._anchor_18 = this.renderer.createTemplateAnchor(this._el_5,(null as any));
    this._vc_18 = new import12.ViewContainer(18,5,this,this._anchor_18);
    this._TemplateRef_18_5 = new import22.TemplateRef_(this,18,this._anchor_18);
    this._NgIf_18_6 = new import17.Wrapper_NgIf(this._vc_18.vcRef,this._TemplateRef_18_5);
    this._text_19 = this.renderer.createText(this._el_5,'\n        ',(null as any));
    this._text_20 = this.renderer.createText(this._el_3,'\n        ',(null as any));
    this._el_21 = import3.createRenderElement(this.renderer,this._el_3,'tr',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_22 = this.renderer.createText(this._el_21,'\n          ',(null as any));
    this._el_23 = import3.createRenderElement(this.renderer,this._el_21,'td',new import3.InlineArray2(2,'class','form-group'),(null as any));
    this._NgClass_23_3 = new import11.Wrapper_NgClass(this.parentView.injectorGet(import13.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import14.KeyValueDiffers,this.parentIndex),new import15.ElementRef(this._el_23),this.renderer);
    this._text_24 = this.renderer.createText(this._el_23,'\n            ',(null as any));
    this._el_25 = import3.createRenderElement(this.renderer,this._el_23,'input',new import3.InlineArray8(8,'class','form-control text-center','maxlength','2','style','width:50px;','type','text'),(null as any));
    this._DefaultValueAccessor_25_3 = new import18.Wrapper_DefaultValueAccessor(this.renderer,new import15.ElementRef(this._el_25));
    this._MaxLengthValidator_25_4 = new import19.Wrapper_MaxLengthValidator();
    this._NG_VALIDATORS_25_5 = [this._MaxLengthValidator_25_4.context];
    this._NG_VALUE_ACCESSOR_25_6 = [this._DefaultValueAccessor_25_3.context];
    this._NgModel_25_7 = new import20.Wrapper_NgModel((null as any),this._NG_VALIDATORS_25_5,(null as any),this._NG_VALUE_ACCESSOR_25_6);
    this._NgControl_25_8 = this._NgModel_25_7.context;
    this._NgControlStatus_25_9 = new import21.Wrapper_NgControlStatus(this._NgControl_25_8);
    this._text_26 = this.renderer.createText(this._el_23,'\n          ',(null as any));
    this._text_27 = this.renderer.createText(this._el_21,'\n          ',(null as any));
    this._el_28 = import3.createRenderElement(this.renderer,this._el_21,'td',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_29 = this.renderer.createText(this._el_28,':',(null as any));
    this._text_30 = this.renderer.createText(this._el_21,'\n          ',(null as any));
    this._el_31 = import3.createRenderElement(this.renderer,this._el_21,'td',new import3.InlineArray2(2,'class','form-group'),(null as any));
    this._NgClass_31_3 = new import11.Wrapper_NgClass(this.parentView.injectorGet(import13.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import14.KeyValueDiffers,this.parentIndex),new import15.ElementRef(this._el_31),this.renderer);
    this._text_32 = this.renderer.createText(this._el_31,'\n            ',(null as any));
    this._el_33 = import3.createRenderElement(this.renderer,this._el_31,'input',new import3.InlineArray8(8,'class','form-control text-center','maxlength','2','style','width:50px;','type','text'),(null as any));
    this._DefaultValueAccessor_33_3 = new import18.Wrapper_DefaultValueAccessor(this.renderer,new import15.ElementRef(this._el_33));
    this._MaxLengthValidator_33_4 = new import19.Wrapper_MaxLengthValidator();
    this._NG_VALIDATORS_33_5 = [this._MaxLengthValidator_33_4.context];
    this._NG_VALUE_ACCESSOR_33_6 = [this._DefaultValueAccessor_33_3.context];
    this._NgModel_33_7 = new import20.Wrapper_NgModel((null as any),this._NG_VALIDATORS_33_5,(null as any),this._NG_VALUE_ACCESSOR_33_6);
    this._NgControl_33_8 = this._NgModel_33_7.context;
    this._NgControlStatus_33_9 = new import21.Wrapper_NgControlStatus(this._NgControl_33_8);
    this._text_34 = this.renderer.createText(this._el_31,'\n          ',(null as any));
    this._text_35 = this.renderer.createText(this._el_21,'\n          ',(null as any));
    this._anchor_36 = this.renderer.createTemplateAnchor(this._el_21,(null as any));
    this._vc_36 = new import12.ViewContainer(36,21,this,this._anchor_36);
    this._TemplateRef_36_5 = new import22.TemplateRef_(this,36,this._anchor_36);
    this._NgIf_36_6 = new import17.Wrapper_NgIf(this._vc_36.vcRef,this._TemplateRef_36_5);
    this._text_37 = this.renderer.createText(this._el_21,'\n        ',(null as any));
    this._text_38 = this.renderer.createText(this._el_3,'\n        ',(null as any));
    this._el_39 = import3.createRenderElement(this.renderer,this._el_3,'tr',new import3.InlineArray2(2,'class','text-center'),(null as any));
    this._NgClass_39_3 = new import11.Wrapper_NgClass(this.parentView.injectorGet(import13.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import14.KeyValueDiffers,this.parentIndex),new import15.ElementRef(this._el_39),this.renderer);
    this._text_40 = this.renderer.createText(this._el_39,'\n          ',(null as any));
    this._el_41 = import3.createRenderElement(this.renderer,this._el_39,'td',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._el_42 = import3.createRenderElement(this.renderer,this._el_41,'a',new import3.InlineArray2(2,'class','btn btn-link'),(null as any));
    this._NgClass_42_3 = new import11.Wrapper_NgClass(this.parentView.injectorGet(import13.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import14.KeyValueDiffers,this.parentIndex),new import15.ElementRef(this._el_42),this.renderer);
    this._el_43 = import3.createRenderElement(this.renderer,this._el_42,'span',new import3.InlineArray2(2,'class','glyphicon glyphicon-chevron-down'),(null as any));
    this._text_44 = this.renderer.createText(this._el_39,'\n          ',(null as any));
    this._el_45 = import3.createRenderElement(this.renderer,this._el_39,'td',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_46 = this.renderer.createText(this._el_45,' ',(null as any));
    this._text_47 = this.renderer.createText(this._el_39,'\n          ',(null as any));
    this._el_48 = import3.createRenderElement(this.renderer,this._el_39,'td',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._el_49 = import3.createRenderElement(this.renderer,this._el_48,'a',new import3.InlineArray2(2,'class','btn btn-link'),(null as any));
    this._NgClass_49_3 = new import11.Wrapper_NgClass(this.parentView.injectorGet(import13.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import14.KeyValueDiffers,this.parentIndex),new import15.ElementRef(this._el_49),this.renderer);
    this._el_50 = import3.createRenderElement(this.renderer,this._el_49,'span',new import3.InlineArray2(2,'class','glyphicon glyphicon-chevron-down'),(null as any));
    this._text_51 = this.renderer.createText(this._el_39,'\n          ',(null as any));
    this._anchor_52 = this.renderer.createTemplateAnchor(this._el_39,(null as any));
    this._vc_52 = new import12.ViewContainer(52,39,this,this._anchor_52);
    this._TemplateRef_52_5 = new import22.TemplateRef_(this,52,this._anchor_52);
    this._NgIf_52_6 = new import17.Wrapper_NgIf(this._vc_52.vcRef,this._TemplateRef_52_5);
    this._text_53 = this.renderer.createText(this._el_39,'\n        ',(null as any));
    this._text_54 = this.renderer.createText(this._el_3,'\n      ',(null as any));
    this._text_55 = this.renderer.createText(this._el_1,'\n    ',(null as any));
    this._text_56 = this.renderer.createText(parentRenderNode,'\n  ',(null as any));
    var disposable_0:Function = import3.subscribeToRenderElement(this,this._el_8,new import3.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_8));
    var disposable_1:Function = import3.subscribeToRenderElement(this,this._el_15,new import3.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_15));
    var disposable_2:Function = import3.subscribeToRenderElement(this,this._el_25,new import3.InlineArray8(8,'ngModelChange',(null as any),'change',(null as any),'blur',(null as any),'input',(null as any)),this.eventHandler(this.handleEvent_25));
    this._NgModel_25_7.subscribe(this,this.eventHandler(this.handleEvent_25),true);
    var disposable_3:Function = import3.subscribeToRenderElement(this,this._el_33,new import3.InlineArray8(8,'ngModelChange',(null as any),'change',(null as any),'blur',(null as any),'input',(null as any)),this.eventHandler(this.handleEvent_33));
    this._NgModel_33_7.subscribe(this,this.eventHandler(this.handleEvent_33),true);
    var disposable_4:Function = import3.subscribeToRenderElement(this,this._el_42,new import3.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_42));
    var disposable_5:Function = import3.subscribeToRenderElement(this,this._el_49,new import3.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_49));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._el_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._el_7,
      this._el_8,
      this._el_9,
      this._text_10,
      this._el_11,
      this._text_12,
      this._text_13,
      this._el_14,
      this._el_15,
      this._el_16,
      this._text_17,
      this._anchor_18,
      this._text_19,
      this._text_20,
      this._el_21,
      this._text_22,
      this._el_23,
      this._text_24,
      this._el_25,
      this._text_26,
      this._text_27,
      this._el_28,
      this._text_29,
      this._text_30,
      this._el_31,
      this._text_32,
      this._el_33,
      this._text_34,
      this._text_35,
      this._anchor_36,
      this._text_37,
      this._text_38,
      this._el_39,
      this._text_40,
      this._el_41,
      this._el_42,
      this._el_43,
      this._text_44,
      this._el_45,
      this._text_46,
      this._text_47,
      this._el_48,
      this._el_49,
      this._el_50,
      this._text_51,
      this._anchor_52,
      this._text_53,
      this._text_54,
      this._text_55,
      this._text_56
    ]
    ),[
      disposable_0,
      disposable_1,
      disposable_2,
      disposable_3,
      disposable_4,
      disposable_5
    ]
    );
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import16.NgClass) && ((8 <= requestNodeIndex) && (requestNodeIndex <= 9)))) { return this._NgClass_8_3.context; }
    if (((token === import16.NgClass) && ((15 <= requestNodeIndex) && (requestNodeIndex <= 16)))) { return this._NgClass_15_3.context; }
    if (((token === import22.TemplateRef) && (18 === requestNodeIndex))) { return this._TemplateRef_18_5; }
    if (((token === import23.NgIf) && (18 === requestNodeIndex))) { return this._NgIf_18_6.context; }
    if (((token === import16.NgClass) && ((5 <= requestNodeIndex) && (requestNodeIndex <= 19)))) { return this._NgClass_5_3.context; }
    if (((token === import24.DefaultValueAccessor) && (25 === requestNodeIndex))) { return this._DefaultValueAccessor_25_3.context; }
    if (((token === import25.MaxLengthValidator) && (25 === requestNodeIndex))) { return this._MaxLengthValidator_25_4.context; }
    if (((token === import26.NG_VALIDATORS) && (25 === requestNodeIndex))) { return this._NG_VALIDATORS_25_5; }
    if (((token === import10.NG_VALUE_ACCESSOR) && (25 === requestNodeIndex))) { return this._NG_VALUE_ACCESSOR_25_6; }
    if (((token === import27.NgModel) && (25 === requestNodeIndex))) { return this._NgModel_25_7.context; }
    if (((token === import28.NgControl) && (25 === requestNodeIndex))) { return this._NgControl_25_8; }
    if (((token === import29.NgControlStatus) && (25 === requestNodeIndex))) { return this._NgControlStatus_25_9.context; }
    if (((token === import16.NgClass) && ((23 <= requestNodeIndex) && (requestNodeIndex <= 26)))) { return this._NgClass_23_3.context; }
    if (((token === import24.DefaultValueAccessor) && (33 === requestNodeIndex))) { return this._DefaultValueAccessor_33_3.context; }
    if (((token === import25.MaxLengthValidator) && (33 === requestNodeIndex))) { return this._MaxLengthValidator_33_4.context; }
    if (((token === import26.NG_VALIDATORS) && (33 === requestNodeIndex))) { return this._NG_VALIDATORS_33_5; }
    if (((token === import10.NG_VALUE_ACCESSOR) && (33 === requestNodeIndex))) { return this._NG_VALUE_ACCESSOR_33_6; }
    if (((token === import27.NgModel) && (33 === requestNodeIndex))) { return this._NgModel_33_7.context; }
    if (((token === import28.NgControl) && (33 === requestNodeIndex))) { return this._NgControl_33_8; }
    if (((token === import29.NgControlStatus) && (33 === requestNodeIndex))) { return this._NgControlStatus_33_9.context; }
    if (((token === import16.NgClass) && ((31 <= requestNodeIndex) && (requestNodeIndex <= 34)))) { return this._NgClass_31_3.context; }
    if (((token === import22.TemplateRef) && (36 === requestNodeIndex))) { return this._TemplateRef_36_5; }
    if (((token === import23.NgIf) && (36 === requestNodeIndex))) { return this._NgIf_36_6.context; }
    if (((token === import16.NgClass) && ((42 <= requestNodeIndex) && (requestNodeIndex <= 43)))) { return this._NgClass_42_3.context; }
    if (((token === import16.NgClass) && ((49 <= requestNodeIndex) && (requestNodeIndex <= 50)))) { return this._NgClass_49_3.context; }
    if (((token === import22.TemplateRef) && (52 === requestNodeIndex))) { return this._TemplateRef_52_5; }
    if (((token === import23.NgIf) && (52 === requestNodeIndex))) { return this._NgIf_52_6.context; }
    if (((token === import16.NgClass) && ((39 <= requestNodeIndex) && (requestNodeIndex <= 53)))) { return this._NgClass_39_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_5_0_0:any = 'text-center';
    this._NgClass_5_3.check_klass(currVal_5_0_0,throwOnChange,false);
    const currVal_5_0_1:any = this._map_88((!this.context.showSpinners || this.context.readonlyInput));
    this._NgClass_5_3.check_ngClass(currVal_5_0_1,throwOnChange,false);
    this._NgClass_5_3.ngDoCheck(this,this._el_5,throwOnChange);
    const currVal_8_0_0:any = 'btn btn-link';
    this._NgClass_8_3.check_klass(currVal_8_0_0,throwOnChange,false);
    const currVal_8_0_1:any = this._map_89(this.context.noIncrementHours());
    this._NgClass_8_3.check_ngClass(currVal_8_0_1,throwOnChange,false);
    this._NgClass_8_3.ngDoCheck(this,this._el_8,throwOnChange);
    const currVal_15_0_0:any = 'btn btn-link';
    this._NgClass_15_3.check_klass(currVal_15_0_0,throwOnChange,false);
    const currVal_15_0_1:any = this._map_90(this.context.noIncrementMinutes());
    this._NgClass_15_3.check_ngClass(currVal_15_0_1,throwOnChange,false);
    this._NgClass_15_3.ngDoCheck(this,this._el_15,throwOnChange);
    const currVal_18_0_0:any = this.context.showMeridian;
    this._NgIf_18_6.check_ngIf(currVal_18_0_0,throwOnChange,false);
    this._NgIf_18_6.ngDoCheck(this,this._anchor_18,throwOnChange);
    const currVal_23_0_0:any = 'form-group';
    this._NgClass_23_3.check_klass(currVal_23_0_0,throwOnChange,false);
    const currVal_23_0_1:any = this._map_91(this.context.invalidHours);
    this._NgClass_23_3.check_ngClass(currVal_23_0_1,throwOnChange,false);
    this._NgClass_23_3.ngDoCheck(this,this._el_23,throwOnChange);
    this._DefaultValueAccessor_25_3.ngDoCheck(this,this._el_25,throwOnChange);
    const currVal_25_1_0:any = '2';
    this._MaxLengthValidator_25_4.check_maxlength(currVal_25_1_0,throwOnChange,false);
    this._MaxLengthValidator_25_4.ngDoCheck(this,this._el_25,throwOnChange);
    const currVal_25_2_0:any = this.context.hours;
    this._NgModel_25_7.check_model(currVal_25_2_0,throwOnChange,false);
    this._NgModel_25_7.ngDoCheck(this,this._el_25,throwOnChange);
    this._NgControlStatus_25_9.ngDoCheck(this,this._el_25,throwOnChange);
    const currVal_31_0_0:any = 'form-group';
    this._NgClass_31_3.check_klass(currVal_31_0_0,throwOnChange,false);
    const currVal_31_0_1:any = this._map_93(this.context.invalidMinutes);
    this._NgClass_31_3.check_ngClass(currVal_31_0_1,throwOnChange,false);
    this._NgClass_31_3.ngDoCheck(this,this._el_31,throwOnChange);
    this._DefaultValueAccessor_33_3.ngDoCheck(this,this._el_33,throwOnChange);
    const currVal_33_1_0:any = '2';
    this._MaxLengthValidator_33_4.check_maxlength(currVal_33_1_0,throwOnChange,false);
    this._MaxLengthValidator_33_4.ngDoCheck(this,this._el_33,throwOnChange);
    const currVal_33_2_0:any = this.context.minutes;
    this._NgModel_33_7.check_model(currVal_33_2_0,throwOnChange,false);
    this._NgModel_33_7.ngDoCheck(this,this._el_33,throwOnChange);
    this._NgControlStatus_33_9.ngDoCheck(this,this._el_33,throwOnChange);
    const currVal_36_0_0:any = this.context.showMeridian;
    this._NgIf_36_6.check_ngIf(currVal_36_0_0,throwOnChange,false);
    this._NgIf_36_6.ngDoCheck(this,this._anchor_36,throwOnChange);
    const currVal_39_0_0:any = 'text-center';
    this._NgClass_39_3.check_klass(currVal_39_0_0,throwOnChange,false);
    const currVal_39_0_1:any = this._map_95((!this.context.showSpinners || this.context.readonlyInput));
    this._NgClass_39_3.check_ngClass(currVal_39_0_1,throwOnChange,false);
    this._NgClass_39_3.ngDoCheck(this,this._el_39,throwOnChange);
    const currVal_42_0_0:any = 'btn btn-link';
    this._NgClass_42_3.check_klass(currVal_42_0_0,throwOnChange,false);
    const currVal_42_0_1:any = this._map_96(this.context.noDecrementHours());
    this._NgClass_42_3.check_ngClass(currVal_42_0_1,throwOnChange,false);
    this._NgClass_42_3.ngDoCheck(this,this._el_42,throwOnChange);
    const currVal_49_0_0:any = 'btn btn-link';
    this._NgClass_49_3.check_klass(currVal_49_0_0,throwOnChange,false);
    const currVal_49_0_1:any = this._map_97(this.context.noDecrementMinutes());
    this._NgClass_49_3.check_ngClass(currVal_49_0_1,throwOnChange,false);
    this._NgClass_49_3.ngDoCheck(this,this._el_49,throwOnChange);
    const currVal_52_0_0:any = this.context.showMeridian;
    this._NgIf_52_6.check_ngIf(currVal_52_0_0,throwOnChange,false);
    this._NgIf_52_6.ngDoCheck(this,this._anchor_52,throwOnChange);
    this._vc_18.detectChangesInNestedViews(throwOnChange);
    this._vc_36.detectChangesInNestedViews(throwOnChange);
    this._vc_52.detectChangesInNestedViews(throwOnChange);
    const currVal_92:any = this.context.readonlyInput;
    if (import3.checkBinding(throwOnChange,this._expr_92,currVal_92)) {
      this.renderer.setElementProperty(this._el_25,'readOnly',currVal_92);
      this._expr_92 = currVal_92;
    }
    this._MaxLengthValidator_25_4.checkHost(this,this,this._el_25,throwOnChange);
    this._NgControlStatus_25_9.checkHost(this,this,this._el_25,throwOnChange);
    const currVal_94:any = this.context.readonlyInput;
    if (import3.checkBinding(throwOnChange,this._expr_94,currVal_94)) {
      this.renderer.setElementProperty(this._el_33,'readOnly',currVal_94);
      this._expr_94 = currVal_94;
    }
    this._MaxLengthValidator_33_4.checkHost(this,this,this._el_33,throwOnChange);
    this._NgControlStatus_33_9.checkHost(this,this,this._el_33,throwOnChange);
  }
  destroyInternal():void {
    this._vc_18.destroyNestedViews();
    this._vc_36.destroyNestedViews();
    this._vc_52.destroyNestedViews();
    this._NgModel_25_7.ngOnDestroy();
    this._NgModel_33_7.ngOnDestroy();
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 18)) { return new View_TimepickerComponent1(this.viewUtils,this,18,this._anchor_18,this._vc_18); }
    if ((nodeIndex == 36)) { return new View_TimepickerComponent2(this.viewUtils,this,36,this._anchor_36,this._vc_36); }
    if ((nodeIndex == 52)) { return new View_TimepickerComponent3(this.viewUtils,this,52,this._anchor_52,this._vc_52); }
    return (null as any);
  }
  handleEvent_8(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.incrementHours()) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_15(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.incrementMinutes()) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_25(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._DefaultValueAccessor_25_3.handleEvent(eventName,$event) && result);
    if ((eventName == 'ngModelChange')) {
      const pd_sub_0:any = ((<any>(this.context.hours = $event)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'change')) {
      const pd_sub_1:any = ((<any>this.context.updateHours()) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'blur')) {
      const pd_sub_2:any = ((<any>this.context.hoursOnBlur()) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
  handleEvent_33(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    result = (this._DefaultValueAccessor_33_3.handleEvent(eventName,$event) && result);
    if ((eventName == 'ngModelChange')) {
      const pd_sub_0:any = ((<any>(this.context.minutes = $event)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'change')) {
      const pd_sub_1:any = ((<any>this.context.updateMinutes()) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'blur')) {
      const pd_sub_2:any = ((<any>this.context.minutesOnBlur()) !== false);
      result = (pd_sub_2 && result);
    }
    return result;
  }
  handleEvent_42(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.decrementHours()) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_49(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.context.decrementMinutes()) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
