// ng2-bootstrap/index.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/linker/ng_module_factory';
import * as import1 from 'ng2-bootstrap/index';
import * as import2 from '@angular/common/src/common_module';
import * as import3 from 'ng2-bootstrap/collapse/collapse.module';
import * as import4 from 'ng2-bootstrap/accordion/accordion.module';
import * as import5 from 'ng2-bootstrap/alert/alert.module';
import * as import6 from 'ng2-bootstrap/buttons/buttons.module';
import * as import7 from 'ng2-bootstrap/carousel/carousel.module';
import * as import8 from '@angular/forms/src/directives';
import * as import9 from '@angular/forms/src/form_providers';
import * as import10 from 'ng2-bootstrap/datepicker/datepicker.module';
import * as import11 from 'ng2-bootstrap/dropdown/dropdown.module';
import * as import12 from 'ng2-bootstrap/modal/modal.module';
import * as import13 from 'ng2-bootstrap/pagination/pagination.module';
import * as import14 from 'ng2-bootstrap/progressbar/progressbar.module';
import * as import15 from 'ng2-bootstrap/popover/popover.module';
import * as import16 from 'ng2-bootstrap/rating/rating.module';
import * as import17 from 'ng2-bootstrap/tabs/tabs.module';
import * as import18 from 'ng2-bootstrap/timepicker/timepicker.module';
import * as import19 from 'ng2-bootstrap/tooltip/tooltip.module';
import * as import20 from 'ng2-bootstrap/typeahead/typeahead.module';
import * as import21 from 'ng2-bootstrap/sortable/sortable.module';
import * as import22 from '@angular/common/src/localization';
import * as import23 from 'ng2-bootstrap/carousel/carousel.config';
import * as import24 from '@angular/forms/src/directives/radio_control_value_accessor';
import * as import25 from 'ng2-bootstrap/accordion/accordion.config';
import * as import26 from 'ng2-bootstrap/alert/alert.config';
import * as import27 from 'ng2-bootstrap/datepicker/datepicker.config';
import * as import28 from 'ng2-bootstrap/dropdown/dropdown.config';
import * as import29 from 'ng2-bootstrap/positioning/positioning.service';
import * as import30 from 'ng2-bootstrap/component-loader/component-loader.factory';
import * as import31 from 'ng2-bootstrap/pagination/pagination.config';
import * as import32 from 'ng2-bootstrap/progressbar/progressbar.config';
import * as import33 from 'ng2-bootstrap/popover/popover.config';
import * as import34 from 'ng2-bootstrap/tabs/tabset.config';
import * as import35 from 'ng2-bootstrap/timepicker/timepicker.config';
import * as import36 from 'ng2-bootstrap/tooltip/tooltip.config';
import * as import37 from 'ng2-bootstrap/sortable/draggable-item.service';
import * as import38 from '@angular/core/src/di/injector';
import * as import39 from './alert/alert.component.ngfactory';
import * as import40 from './datepicker/datepicker.component.ngfactory';
import * as import41 from './modal/modal-backdrop.component.ngfactory';
import * as import42 from './popover/popover-container.component.ngfactory';
import * as import43 from './tooltip/tooltip-container.component.ngfactory';
import * as import44 from './typeahead/typeahead-container.component.ngfactory';
import * as import45 from '@angular/core/src/i18n/tokens';
import * as import46 from '@angular/core/src/zone/ng_zone';
class BsRootModuleInjector extends import0.NgModuleInjector<import1.BsRootModule> {
  _CommonModule_0:import2.CommonModule;
  _CollapseModule_1:import3.CollapseModule;
  _AccordionModule_2:import4.AccordionModule;
  _AlertModule_3:import5.AlertModule;
  _ButtonsModule_4:import6.ButtonsModule;
  _CarouselModule_5:import7.CarouselModule;
  _InternalFormsSharedModule_6:import8.InternalFormsSharedModule;
  _FormsModule_7:import9.FormsModule;
  _DatepickerModule_8:import10.DatepickerModule;
  _DropdownModule_9:import11.DropdownModule;
  _ModalModule_10:import12.ModalModule;
  _PaginationModule_11:import13.PaginationModule;
  _ProgressbarModule_12:import14.ProgressbarModule;
  _PopoverModule_13:import15.PopoverModule;
  _RatingModule_14:import16.RatingModule;
  _TabsModule_15:import17.TabsModule;
  _TimepickerModule_16:import18.TimepickerModule;
  _TooltipModule_17:import19.TooltipModule;
  _TypeaheadModule_18:import20.TypeaheadModule;
  _SortableModule_19:import21.SortableModule;
  _BsRootModule_20:import1.BsRootModule;
  __NgLocalization_21:import22.NgLocaleLocalization;
  __CarouselConfig_22:import23.CarouselConfig;
  __RadioControlRegistry_23:import24.RadioControlRegistry;
  __AccordionConfig_24:import25.AccordionConfig;
  __AlertConfig_25:import26.AlertConfig;
  __DatepickerConfig_26:import27.DatepickerConfig;
  __DropdownConfig_27:import28.DropdownConfig;
  __PositioningService_28:import29.PositioningService;
  __ComponentLoaderFactory_29:import30.ComponentLoaderFactory;
  __PaginationConfig_30:import31.PaginationConfig;
  __ProgressbarConfig_31:import32.ProgressbarConfig;
  __PopoverConfig_32:import33.PopoverConfig;
  __TabsetConfig_33:import34.TabsetConfig;
  __TimepickerConfig_34:import35.TimepickerConfig;
  __TooltipConfig_35:import36.TooltipConfig;
  __DraggableItemService_36:import37.DraggableItemService;
  constructor(parent:import38.Injector) {
    super(parent,[
      import39.AlertComponentNgFactory,
      import40.DatePickerComponentNgFactory,
      import41.ModalBackdropComponentNgFactory,
      import42.PopoverContainerComponentNgFactory,
      import43.TooltipContainerComponentNgFactory,
      import44.TypeaheadContainerComponentNgFactory
    ]
    ,([] as any[]));
  }
  get _NgLocalization_21():import22.NgLocaleLocalization {
    if ((this.__NgLocalization_21 == null)) { (this.__NgLocalization_21 = new import22.NgLocaleLocalization(this.parent.get(import45.LOCALE_ID))); }
    return this.__NgLocalization_21;
  }
  get _CarouselConfig_22():import23.CarouselConfig {
    if ((this.__CarouselConfig_22 == null)) { (this.__CarouselConfig_22 = new import23.CarouselConfig()); }
    return this.__CarouselConfig_22;
  }
  get _RadioControlRegistry_23():import24.RadioControlRegistry {
    if ((this.__RadioControlRegistry_23 == null)) { (this.__RadioControlRegistry_23 = new import24.RadioControlRegistry()); }
    return this.__RadioControlRegistry_23;
  }
  get _AccordionConfig_24():import25.AccordionConfig {
    if ((this.__AccordionConfig_24 == null)) { (this.__AccordionConfig_24 = new import25.AccordionConfig()); }
    return this.__AccordionConfig_24;
  }
  get _AlertConfig_25():import26.AlertConfig {
    if ((this.__AlertConfig_25 == null)) { (this.__AlertConfig_25 = new import26.AlertConfig()); }
    return this.__AlertConfig_25;
  }
  get _DatepickerConfig_26():import27.DatepickerConfig {
    if ((this.__DatepickerConfig_26 == null)) { (this.__DatepickerConfig_26 = new import27.DatepickerConfig()); }
    return this.__DatepickerConfig_26;
  }
  get _DropdownConfig_27():import28.DropdownConfig {
    if ((this.__DropdownConfig_27 == null)) { (this.__DropdownConfig_27 = new import28.DropdownConfig()); }
    return this.__DropdownConfig_27;
  }
  get _PositioningService_28():import29.PositioningService {
    if ((this.__PositioningService_28 == null)) { (this.__PositioningService_28 = new import29.PositioningService()); }
    return this.__PositioningService_28;
  }
  get _ComponentLoaderFactory_29():import30.ComponentLoaderFactory {
    if ((this.__ComponentLoaderFactory_29 == null)) { (this.__ComponentLoaderFactory_29 = new import30.ComponentLoaderFactory(this,this.parent.get(import46.NgZone),this,this._PositioningService_28)); }
    return this.__ComponentLoaderFactory_29;
  }
  get _PaginationConfig_30():import31.PaginationConfig {
    if ((this.__PaginationConfig_30 == null)) { (this.__PaginationConfig_30 = new import31.PaginationConfig()); }
    return this.__PaginationConfig_30;
  }
  get _ProgressbarConfig_31():import32.ProgressbarConfig {
    if ((this.__ProgressbarConfig_31 == null)) { (this.__ProgressbarConfig_31 = new import32.ProgressbarConfig()); }
    return this.__ProgressbarConfig_31;
  }
  get _PopoverConfig_32():import33.PopoverConfig {
    if ((this.__PopoverConfig_32 == null)) { (this.__PopoverConfig_32 = new import33.PopoverConfig()); }
    return this.__PopoverConfig_32;
  }
  get _TabsetConfig_33():import34.TabsetConfig {
    if ((this.__TabsetConfig_33 == null)) { (this.__TabsetConfig_33 = new import34.TabsetConfig()); }
    return this.__TabsetConfig_33;
  }
  get _TimepickerConfig_34():import35.TimepickerConfig {
    if ((this.__TimepickerConfig_34 == null)) { (this.__TimepickerConfig_34 = new import35.TimepickerConfig()); }
    return this.__TimepickerConfig_34;
  }
  get _TooltipConfig_35():import36.TooltipConfig {
    if ((this.__TooltipConfig_35 == null)) { (this.__TooltipConfig_35 = new import36.TooltipConfig()); }
    return this.__TooltipConfig_35;
  }
  get _DraggableItemService_36():import37.DraggableItemService {
    if ((this.__DraggableItemService_36 == null)) { (this.__DraggableItemService_36 = new import37.DraggableItemService()); }
    return this.__DraggableItemService_36;
  }
  createInternal():import1.BsRootModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._CollapseModule_1 = new import3.CollapseModule();
    this._AccordionModule_2 = new import4.AccordionModule();
    this._AlertModule_3 = new import5.AlertModule();
    this._ButtonsModule_4 = new import6.ButtonsModule();
    this._CarouselModule_5 = new import7.CarouselModule();
    this._InternalFormsSharedModule_6 = new import8.InternalFormsSharedModule();
    this._FormsModule_7 = new import9.FormsModule();
    this._DatepickerModule_8 = new import10.DatepickerModule();
    this._DropdownModule_9 = new import11.DropdownModule();
    this._ModalModule_10 = new import12.ModalModule();
    this._PaginationModule_11 = new import13.PaginationModule();
    this._ProgressbarModule_12 = new import14.ProgressbarModule();
    this._PopoverModule_13 = new import15.PopoverModule();
    this._RatingModule_14 = new import16.RatingModule();
    this._TabsModule_15 = new import17.TabsModule();
    this._TimepickerModule_16 = new import18.TimepickerModule();
    this._TooltipModule_17 = new import19.TooltipModule();
    this._TypeaheadModule_18 = new import20.TypeaheadModule();
    this._SortableModule_19 = new import21.SortableModule();
    this._BsRootModule_20 = new import1.BsRootModule();
    return this._BsRootModule_20;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.CollapseModule)) { return this._CollapseModule_1; }
    if ((token === import4.AccordionModule)) { return this._AccordionModule_2; }
    if ((token === import5.AlertModule)) { return this._AlertModule_3; }
    if ((token === import6.ButtonsModule)) { return this._ButtonsModule_4; }
    if ((token === import7.CarouselModule)) { return this._CarouselModule_5; }
    if ((token === import8.InternalFormsSharedModule)) { return this._InternalFormsSharedModule_6; }
    if ((token === import9.FormsModule)) { return this._FormsModule_7; }
    if ((token === import10.DatepickerModule)) { return this._DatepickerModule_8; }
    if ((token === import11.DropdownModule)) { return this._DropdownModule_9; }
    if ((token === import12.ModalModule)) { return this._ModalModule_10; }
    if ((token === import13.PaginationModule)) { return this._PaginationModule_11; }
    if ((token === import14.ProgressbarModule)) { return this._ProgressbarModule_12; }
    if ((token === import15.PopoverModule)) { return this._PopoverModule_13; }
    if ((token === import16.RatingModule)) { return this._RatingModule_14; }
    if ((token === import17.TabsModule)) { return this._TabsModule_15; }
    if ((token === import18.TimepickerModule)) { return this._TimepickerModule_16; }
    if ((token === import19.TooltipModule)) { return this._TooltipModule_17; }
    if ((token === import20.TypeaheadModule)) { return this._TypeaheadModule_18; }
    if ((token === import21.SortableModule)) { return this._SortableModule_19; }
    if ((token === import1.BsRootModule)) { return this._BsRootModule_20; }
    if ((token === import22.NgLocalization)) { return this._NgLocalization_21; }
    if ((token === import23.CarouselConfig)) { return this._CarouselConfig_22; }
    if ((token === import24.RadioControlRegistry)) { return this._RadioControlRegistry_23; }
    if ((token === import25.AccordionConfig)) { return this._AccordionConfig_24; }
    if ((token === import26.AlertConfig)) { return this._AlertConfig_25; }
    if ((token === import27.DatepickerConfig)) { return this._DatepickerConfig_26; }
    if ((token === import28.DropdownConfig)) { return this._DropdownConfig_27; }
    if ((token === import29.PositioningService)) { return this._PositioningService_28; }
    if ((token === import30.ComponentLoaderFactory)) { return this._ComponentLoaderFactory_29; }
    if ((token === import31.PaginationConfig)) { return this._PaginationConfig_30; }
    if ((token === import32.ProgressbarConfig)) { return this._ProgressbarConfig_31; }
    if ((token === import33.PopoverConfig)) { return this._PopoverConfig_32; }
    if ((token === import34.TabsetConfig)) { return this._TabsetConfig_33; }
    if ((token === import35.TimepickerConfig)) { return this._TimepickerConfig_34; }
    if ((token === import36.TooltipConfig)) { return this._TooltipConfig_35; }
    if ((token === import37.DraggableItemService)) { return this._DraggableItemService_36; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const BsRootModuleNgFactory:import0.NgModuleFactory<import1.BsRootModule> = new import0.NgModuleFactory(BsRootModuleInjector,import1.BsRootModule);
class Ng2BootstrapModuleInjector extends import0.NgModuleInjector<import1.Ng2BootstrapModule> {
  _CommonModule_0:import2.CommonModule;
  _CollapseModule_1:import3.CollapseModule;
  _AccordionModule_2:import4.AccordionModule;
  _AlertModule_3:import5.AlertModule;
  _ButtonsModule_4:import6.ButtonsModule;
  _CarouselModule_5:import7.CarouselModule;
  _InternalFormsSharedModule_6:import8.InternalFormsSharedModule;
  _FormsModule_7:import9.FormsModule;
  _DatepickerModule_8:import10.DatepickerModule;
  _DropdownModule_9:import11.DropdownModule;
  _ModalModule_10:import12.ModalModule;
  _PaginationModule_11:import13.PaginationModule;
  _ProgressbarModule_12:import14.ProgressbarModule;
  _PopoverModule_13:import15.PopoverModule;
  _RatingModule_14:import16.RatingModule;
  _TabsModule_15:import17.TabsModule;
  _TimepickerModule_16:import18.TimepickerModule;
  _TooltipModule_17:import19.TooltipModule;
  _TypeaheadModule_18:import20.TypeaheadModule;
  _Ng2BootstrapModule_19:import1.Ng2BootstrapModule;
  __NgLocalization_20:import22.NgLocaleLocalization;
  __CarouselConfig_21:import23.CarouselConfig;
  __RadioControlRegistry_22:import24.RadioControlRegistry;
  constructor(parent:import38.Injector) {
    super(parent,[
      import39.AlertComponentNgFactory,
      import40.DatePickerComponentNgFactory,
      import41.ModalBackdropComponentNgFactory,
      import42.PopoverContainerComponentNgFactory,
      import43.TooltipContainerComponentNgFactory,
      import44.TypeaheadContainerComponentNgFactory
    ]
    ,([] as any[]));
  }
  get _NgLocalization_20():import22.NgLocaleLocalization {
    if ((this.__NgLocalization_20 == null)) { (this.__NgLocalization_20 = new import22.NgLocaleLocalization(this.parent.get(import45.LOCALE_ID))); }
    return this.__NgLocalization_20;
  }
  get _CarouselConfig_21():import23.CarouselConfig {
    if ((this.__CarouselConfig_21 == null)) { (this.__CarouselConfig_21 = new import23.CarouselConfig()); }
    return this.__CarouselConfig_21;
  }
  get _RadioControlRegistry_22():import24.RadioControlRegistry {
    if ((this.__RadioControlRegistry_22 == null)) { (this.__RadioControlRegistry_22 = new import24.RadioControlRegistry()); }
    return this.__RadioControlRegistry_22;
  }
  createInternal():import1.Ng2BootstrapModule {
    this._CommonModule_0 = new import2.CommonModule();
    this._CollapseModule_1 = new import3.CollapseModule();
    this._AccordionModule_2 = new import4.AccordionModule();
    this._AlertModule_3 = new import5.AlertModule();
    this._ButtonsModule_4 = new import6.ButtonsModule();
    this._CarouselModule_5 = new import7.CarouselModule();
    this._InternalFormsSharedModule_6 = new import8.InternalFormsSharedModule();
    this._FormsModule_7 = new import9.FormsModule();
    this._DatepickerModule_8 = new import10.DatepickerModule();
    this._DropdownModule_9 = new import11.DropdownModule();
    this._ModalModule_10 = new import12.ModalModule();
    this._PaginationModule_11 = new import13.PaginationModule();
    this._ProgressbarModule_12 = new import14.ProgressbarModule();
    this._PopoverModule_13 = new import15.PopoverModule();
    this._RatingModule_14 = new import16.RatingModule();
    this._TabsModule_15 = new import17.TabsModule();
    this._TimepickerModule_16 = new import18.TimepickerModule();
    this._TooltipModule_17 = new import19.TooltipModule();
    this._TypeaheadModule_18 = new import20.TypeaheadModule();
    this._Ng2BootstrapModule_19 = new import1.Ng2BootstrapModule();
    return this._Ng2BootstrapModule_19;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CommonModule)) { return this._CommonModule_0; }
    if ((token === import3.CollapseModule)) { return this._CollapseModule_1; }
    if ((token === import4.AccordionModule)) { return this._AccordionModule_2; }
    if ((token === import5.AlertModule)) { return this._AlertModule_3; }
    if ((token === import6.ButtonsModule)) { return this._ButtonsModule_4; }
    if ((token === import7.CarouselModule)) { return this._CarouselModule_5; }
    if ((token === import8.InternalFormsSharedModule)) { return this._InternalFormsSharedModule_6; }
    if ((token === import9.FormsModule)) { return this._FormsModule_7; }
    if ((token === import10.DatepickerModule)) { return this._DatepickerModule_8; }
    if ((token === import11.DropdownModule)) { return this._DropdownModule_9; }
    if ((token === import12.ModalModule)) { return this._ModalModule_10; }
    if ((token === import13.PaginationModule)) { return this._PaginationModule_11; }
    if ((token === import14.ProgressbarModule)) { return this._ProgressbarModule_12; }
    if ((token === import15.PopoverModule)) { return this._PopoverModule_13; }
    if ((token === import16.RatingModule)) { return this._RatingModule_14; }
    if ((token === import17.TabsModule)) { return this._TabsModule_15; }
    if ((token === import18.TimepickerModule)) { return this._TimepickerModule_16; }
    if ((token === import19.TooltipModule)) { return this._TooltipModule_17; }
    if ((token === import20.TypeaheadModule)) { return this._TypeaheadModule_18; }
    if ((token === import1.Ng2BootstrapModule)) { return this._Ng2BootstrapModule_19; }
    if ((token === import22.NgLocalization)) { return this._NgLocalization_20; }
    if ((token === import23.CarouselConfig)) { return this._CarouselConfig_21; }
    if ((token === import24.RadioControlRegistry)) { return this._RadioControlRegistry_22; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const Ng2BootstrapModuleNgFactory:import0.NgModuleFactory<import1.Ng2BootstrapModule> = new import0.NgModuleFactory(Ng2BootstrapModuleInjector,import1.Ng2BootstrapModule);
