// ng2-bootstrap/datepicker/yearpicker.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from 'ng2-bootstrap/datepicker/yearpicker.component';
import * as import1 from '@angular/core/src/linker/view';
import * as import2 from '@angular/core/src/render/api';
import * as import3 from '@angular/core/src/linker/view_utils';
import * as import4 from '@angular/core/src/metadata/view';
import * as import5 from '@angular/core/src/linker/view_type';
import * as import6 from '@angular/core/src/change_detection/constants';
import * as import7 from '@angular/core/src/linker/component_factory';
import * as import8 from 'ng2-bootstrap/datepicker/datepicker-inner.component';
import * as import9 from '../../@angular/common/src/directives/ng_class.ngfactory';
import * as import10 from '@angular/core/src/linker/view_container';
import * as import11 from '@angular/core/src/change_detection/change_detection_util';
import * as import12 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import13 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import14 from '@angular/core/src/linker/element_ref';
import * as import15 from '@angular/common/src/directives/ng_class';
import * as import16 from '../../@angular/common/src/directives/ng_for.ngfactory';
import * as import17 from '@angular/core/src/linker/template_ref';
import * as import18 from '@angular/common/src/directives/ng_for';
import * as import19 from '../../@angular/common/src/directives/ng_if.ngfactory';
import * as import20 from '@angular/common/src/directives/ng_if';
export class Wrapper_YearPickerComponent {
  /*private*/ _eventHandler:Function;
  context:import0.YearPickerComponent;
  /*private*/ _changed:boolean;
  constructor(p0:any) {
    this._changed = false;
    this.context = new import0.YearPickerComponent(p0);
  }
  ngOnDetach(view:import1.AppView<any>,componentView:import1.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
  }
  ngDoCheck(view:import1.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    if (!throwOnChange) { if ((view.numberOfChecks === 0)) { this.context.ngOnInit(); } }
    return changed;
  }
  checkHost(view:import1.AppView<any>,componentView:import1.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import1.AppView<any>,_eventHandler:any):void {
    this._eventHandler = _eventHandler;
  }
}
var renderType_YearPickerComponent_Host:import2.RenderComponentType = import3.createRenderComponentType('',0,import4.ViewEncapsulation.None,([] as any[]),{});
class View_YearPickerComponent_Host0 extends import1.AppView<any> {
  _el_0:any;
  compView_0:import1.AppView<import0.YearPickerComponent>;
  _YearPickerComponent_0_3:Wrapper_YearPickerComponent;
  constructor(viewUtils:import3.ViewUtils,parentView:import1.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_YearPickerComponent_Host0,renderType_YearPickerComponent_Host,import5.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import6.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer,'yearpicker',import3.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_YearPickerComponent0(this.viewUtils,this,0,this._el_0);
    this._YearPickerComponent_0_3 = new Wrapper_YearPickerComponent(this.injectorGet(import8.DatePickerInnerComponent,this.parentIndex));
    this.compView_0.create(this._YearPickerComponent_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import7.ComponentRef_<any>(0,this,this._el_0,this._YearPickerComponent_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import0.YearPickerComponent) && (0 === requestNodeIndex))) { return this._YearPickerComponent_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._YearPickerComponent_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this.compView_0.internalDetectChanges(throwOnChange);
  }
  destroyInternal():void {
    this.compView_0.destroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const YearPickerComponentNgFactory:import7.ComponentFactory<import0.YearPickerComponent> = new import7.ComponentFactory<import0.YearPickerComponent>('yearpicker',View_YearPickerComponent_Host0,import0.YearPickerComponent);
const styles_YearPickerComponent:any[] = ([] as any[]);
class View_YearPickerComponent3 extends import1.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _NgClass_2_3:import9.Wrapper_NgClass;
  _text_3:any;
  _el_4:any;
  _NgClass_4_3:import9.Wrapper_NgClass;
  _text_5:any;
  _text_6:any;
  _text_7:any;
  /*private*/ _expr_10:any;
  _map_11:any;
  _map_12:any;
  /*private*/ _expr_13:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import1.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import10.ViewContainer) {
    super(View_YearPickerComponent3,renderType_YearPickerComponent,import5.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import6.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_10 = import11.UNINITIALIZED;
    this._map_11 = import3.pureProxy4((p0:any,p1:any,p2:any,p3:any):{[key: string]:any} => {
      return {
        'btn-link': p0,
        'btn-info': p1,
        disabled: p2,
        active: p3
      }
      ;
    });
    this._map_12 = import3.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'text-success': p0,
        'text-info': p1
      }
      ;
    });
    this._expr_13 = import11.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'td',new import3.InlineArray4(4,'class','text-center','role','gridcell'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n        ',(null as any));
    this._el_2 = import3.createRenderElement(this.renderer,this._el_0,'button',new import3.InlineArray8(8,'class','btn btn-default','style','min-width:100%;','tabindex','-1','type','button'),(null as any));
    this._NgClass_2_3 = new import9.Wrapper_NgClass(this.parentView.parentView.parentView.injectorGet(import12.IterableDiffers,this.parentView.parentView.parentIndex),this.parentView.parentView.parentView.injectorGet(import13.KeyValueDiffers,this.parentView.parentView.parentIndex),new import14.ElementRef(this._el_2),this.renderer);
    this._text_3 = this.renderer.createText(this._el_2,'\n          ',(null as any));
    this._el_4 = import3.createRenderElement(this.renderer,this._el_2,'span',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_4_3 = new import9.Wrapper_NgClass(this.parentView.parentView.parentView.injectorGet(import12.IterableDiffers,this.parentView.parentView.parentIndex),this.parentView.parentView.parentView.injectorGet(import13.KeyValueDiffers,this.parentView.parentView.parentIndex),new import14.ElementRef(this._el_4),this.renderer);
    this._text_5 = this.renderer.createText(this._el_4,'',(null as any));
    this._text_6 = this.renderer.createText(this._el_2,'\n        ',(null as any));
    this._text_7 = this.renderer.createText(this._el_0,'\n      ',(null as any));
    var disposable_0:Function = import3.subscribeToRenderElement(this,this._el_2,new import3.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_2));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._el_4,
      this._text_5,
      this._text_6,
      this._text_7
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import15.NgClass) && ((4 <= requestNodeIndex) && (requestNodeIndex <= 5)))) { return this._NgClass_4_3.context; }
    if (((token === import15.NgClass) && ((2 <= requestNodeIndex) && (requestNodeIndex <= 6)))) { return this._NgClass_2_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:any = 'btn btn-default';
    this._NgClass_2_3.check_klass(currVal_2_0_0,throwOnChange,false);
    const currVal_2_0_1:any = this._map_11(((this.parentView.parentView.parentView.context.isBs4 && !this.context.$implicit.selected) && !this.parentView.parentView.parentView.context.datePicker.isActive(this.context.$implicit)),(this.context.$implicit.selected || ((this.parentView.parentView.parentView.context.isBs4 && !this.context.$implicit.selected) && this.parentView.parentView.parentView.context.datePicker.isActive(this.context.$implicit))),this.context.$implicit.disabled,(!this.parentView.parentView.parentView.context.isBs4 && this.parentView.parentView.parentView.context.datePicker.isActive(this.context.$implicit)));
    this._NgClass_2_3.check_ngClass(currVal_2_0_1,throwOnChange,false);
    this._NgClass_2_3.ngDoCheck(this,this._el_2,throwOnChange);
    const currVal_4_0_0:any = this._map_12((this.parentView.parentView.parentView.context.isBs4 && this.context.$implicit.current),(!this.parentView.parentView.parentView.context.isBs4 && this.context.$implicit.current));
    this._NgClass_4_3.check_ngClass(currVal_4_0_0,throwOnChange,false);
    this._NgClass_4_3.ngDoCheck(this,this._el_4,throwOnChange);
    const currVal_10:any = this.context.$implicit.disabled;
    if (import3.checkBinding(throwOnChange,this._expr_10,currVal_10)) {
      this.renderer.setElementProperty(this._el_2,'disabled',currVal_10);
      this._expr_10 = currVal_10;
    }
    const currVal_13:any = import3.inlineInterpolate(1,'',this.context.$implicit.label,'');
    if (import3.checkBinding(throwOnChange,this._expr_13,currVal_13)) {
      this.renderer.setText(this._text_5,currVal_13);
      this._expr_13 = currVal_13;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  handleEvent_2(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.parentView.parentView.context.datePicker.select(this.context.$implicit.date)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_YearPickerComponent2 extends import1.AppView<any> {
  _el_0:any;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import10.ViewContainer;
  _TemplateRef_2_5:any;
  _NgFor_2_6:import16.Wrapper_NgFor;
  _text_3:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import1.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import10.ViewContainer) {
    super(View_YearPickerComponent2,renderType_YearPickerComponent,import5.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import6.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'tr',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n      ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import10.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import17.TemplateRef_(this,2,this._anchor_2);
    this._NgFor_2_6 = new import16.Wrapper_NgFor(this._vc_2.vcRef,this._TemplateRef_2_5,this.parentView.parentView.injectorGet(import12.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.ref);
    this._text_3 = this.renderer.createText(this._el_0,'\n    ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import17.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import18.NgFor) && (2 === requestNodeIndex))) { return this._NgFor_2_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:any = this.context.$implicit;
    this._NgFor_2_6.check_ngForOf(currVal_2_0_0,throwOnChange,false);
    this._NgFor_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import1.AppView<any> {
    if ((nodeIndex == 2)) { return new View_YearPickerComponent3(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
}
class View_YearPickerComponent1 extends import1.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _el_4:any;
  _text_5:any;
  _el_6:any;
  _text_7:any;
  _el_8:any;
  _text_9:any;
  _el_10:any;
  _text_11:any;
  _text_12:any;
  _text_13:any;
  _el_14:any;
  _text_15:any;
  _el_16:any;
  _NgClass_16_3:import9.Wrapper_NgClass;
  _text_17:any;
  _el_18:any;
  _text_19:any;
  _text_20:any;
  _text_21:any;
  _text_22:any;
  _el_23:any;
  _text_24:any;
  _el_25:any;
  _text_26:any;
  _el_27:any;
  _text_28:any;
  _text_29:any;
  _text_30:any;
  _text_31:any;
  _text_32:any;
  _el_33:any;
  _text_34:any;
  _anchor_35:any;
  /*private*/ _vc_35:import10.ViewContainer;
  _TemplateRef_35_5:any;
  _NgFor_35_6:import16.Wrapper_NgFor;
  _text_36:any;
  _text_37:any;
  /*private*/ _expr_42:any;
  /*private*/ _expr_43:any;
  /*private*/ _expr_44:any;
  _map_45:any;
  /*private*/ _expr_46:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import1.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import10.ViewContainer) {
    super(View_YearPickerComponent1,renderType_YearPickerComponent,import5.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import6.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_42 = import11.UNINITIALIZED;
    this._expr_43 = import11.UNINITIALIZED;
    this._expr_44 = import11.UNINITIALIZED;
    this._map_45 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {disabled: p0};
    });
    this._expr_46 = import11.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'table',new import3.InlineArray2(2,'role','grid'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n  ',(null as any));
    this._el_2 = import3.createRenderElement(this.renderer,this._el_0,'thead',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'\n    ',(null as any));
    this._el_4 = import3.createRenderElement(this.renderer,this._el_2,'tr',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_5 = this.renderer.createText(this._el_4,'\n      ',(null as any));
    this._el_6 = import3.createRenderElement(this.renderer,this._el_4,'th',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_7 = this.renderer.createText(this._el_6,'\n        ',(null as any));
    this._el_8 = import3.createRenderElement(this.renderer,this._el_6,'button',new import3.InlineArray8(6,'class','btn btn-default btn-sm pull-left','tabindex','-1','type','button'),(null as any));
    this._text_9 = this.renderer.createText(this._el_8,'\n          ',(null as any));
    this._el_10 = import3.createRenderElement(this.renderer,this._el_8,'i',new import3.InlineArray2(2,'class','glyphicon glyphicon-chevron-left'),(null as any));
    this._text_11 = this.renderer.createText(this._el_8,'\n        ',(null as any));
    this._text_12 = this.renderer.createText(this._el_6,'\n      ',(null as any));
    this._text_13 = this.renderer.createText(this._el_4,'\n      ',(null as any));
    this._el_14 = import3.createRenderElement(this.renderer,this._el_4,'th',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_15 = this.renderer.createText(this._el_14,'\n        ',(null as any));
    this._el_16 = import3.createRenderElement(this.renderer,this._el_14,'button',new import3.InlineArray16(10,'class','btn btn-default btn-sm','role','heading','style','width:100%;','tabindex','-1','type','button'),(null as any));
    this._NgClass_16_3 = new import9.Wrapper_NgClass(this.parentView.injectorGet(import12.IterableDiffers,this.parentIndex),this.parentView.injectorGet(import13.KeyValueDiffers,this.parentIndex),new import14.ElementRef(this._el_16),this.renderer);
    this._text_17 = this.renderer.createText(this._el_16,'\n          ',(null as any));
    this._el_18 = import3.createRenderElement(this.renderer,this._el_16,'strong',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_19 = this.renderer.createText(this._el_18,'',(null as any));
    this._text_20 = this.renderer.createText(this._el_16,'\n        ',(null as any));
    this._text_21 = this.renderer.createText(this._el_14,'\n      ',(null as any));
    this._text_22 = this.renderer.createText(this._el_4,'\n      ',(null as any));
    this._el_23 = import3.createRenderElement(this.renderer,this._el_4,'th',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_24 = this.renderer.createText(this._el_23,'\n        ',(null as any));
    this._el_25 = import3.createRenderElement(this.renderer,this._el_23,'button',new import3.InlineArray8(6,'class','btn btn-default btn-sm pull-right','tabindex','-1','type','button'),(null as any));
    this._text_26 = this.renderer.createText(this._el_25,'\n          ',(null as any));
    this._el_27 = import3.createRenderElement(this.renderer,this._el_25,'i',new import3.InlineArray2(2,'class','glyphicon glyphicon-chevron-right'),(null as any));
    this._text_28 = this.renderer.createText(this._el_25,'\n        ',(null as any));
    this._text_29 = this.renderer.createText(this._el_23,'\n      ',(null as any));
    this._text_30 = this.renderer.createText(this._el_4,'\n    ',(null as any));
    this._text_31 = this.renderer.createText(this._el_2,'\n  ',(null as any));
    this._text_32 = this.renderer.createText(this._el_0,'\n  ',(null as any));
    this._el_33 = import3.createRenderElement(this.renderer,this._el_0,'tbody',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_34 = this.renderer.createText(this._el_33,'\n    ',(null as any));
    this._anchor_35 = this.renderer.createTemplateAnchor(this._el_33,(null as any));
    this._vc_35 = new import10.ViewContainer(35,33,this,this._anchor_35);
    this._TemplateRef_35_5 = new import17.TemplateRef_(this,35,this._anchor_35);
    this._NgFor_35_6 = new import16.Wrapper_NgFor(this._vc_35.vcRef,this._TemplateRef_35_5,this.parentView.injectorGet(import12.IterableDiffers,this.parentIndex),this.parentView.ref);
    this._text_36 = this.renderer.createText(this._el_33,'\n  ',(null as any));
    this._text_37 = this.renderer.createText(this._el_0,'\n',(null as any));
    var disposable_0:Function = import3.subscribeToRenderElement(this,this._el_8,new import3.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_8));
    var disposable_1:Function = import3.subscribeToRenderElement(this,this._el_16,new import3.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_16));
    var disposable_2:Function = import3.subscribeToRenderElement(this,this._el_25,new import3.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_25));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._el_4,
      this._text_5,
      this._el_6,
      this._text_7,
      this._el_8,
      this._text_9,
      this._el_10,
      this._text_11,
      this._text_12,
      this._text_13,
      this._el_14,
      this._text_15,
      this._el_16,
      this._text_17,
      this._el_18,
      this._text_19,
      this._text_20,
      this._text_21,
      this._text_22,
      this._el_23,
      this._text_24,
      this._el_25,
      this._text_26,
      this._el_27,
      this._text_28,
      this._text_29,
      this._text_30,
      this._text_31,
      this._text_32,
      this._el_33,
      this._text_34,
      this._anchor_35,
      this._text_36,
      this._text_37
    ]
    ),[
      disposable_0,
      disposable_1,
      disposable_2
    ]
    );
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import15.NgClass) && ((16 <= requestNodeIndex) && (requestNodeIndex <= 20)))) { return this._NgClass_16_3.context; }
    if (((token === import17.TemplateRef) && (35 === requestNodeIndex))) { return this._TemplateRef_35_5; }
    if (((token === import18.NgFor) && (35 === requestNodeIndex))) { return this._NgFor_35_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_16_0_0:any = 'btn btn-default btn-sm';
    this._NgClass_16_3.check_klass(currVal_16_0_0,throwOnChange,false);
    const currVal_16_0_1:any = this._map_45((this.parentView.context.datePicker.datepickerMode === this.parentView.context.datePicker.maxMode));
    this._NgClass_16_3.check_ngClass(currVal_16_0_1,throwOnChange,false);
    this._NgClass_16_3.ngDoCheck(this,this._el_16,throwOnChange);
    const currVal_35_0_0:any = this.parentView.context.rows;
    this._NgFor_35_6.check_ngForOf(currVal_35_0_0,throwOnChange,false);
    this._NgFor_35_6.ngDoCheck(this,this._anchor_35,throwOnChange);
    this._vc_35.detectChangesInNestedViews(throwOnChange);
    const currVal_42:any = (((this.parentView.context.datePicker.yearColLimit - 2) <= 0)? 1: (this.parentView.context.datePicker.yearColLimit - 2));
    if (import3.checkBinding(throwOnChange,this._expr_42,currVal_42)) {
      this.renderer.setElementAttribute(this._el_14,'colspan',((currVal_42 == null)? (null as any): currVal_42.toString()));
      this._expr_42 = currVal_42;
    }
    const currVal_43:any = (this.parentView.context.datePicker.uniqueId + '-title');
    if (import3.checkBinding(throwOnChange,this._expr_43,currVal_43)) {
      this.renderer.setElementProperty(this._el_16,'id',currVal_43);
      this._expr_43 = currVal_43;
    }
    const currVal_44:any = (this.parentView.context.datePicker.datepickerMode === this.parentView.context.datePicker.maxMode);
    if (import3.checkBinding(throwOnChange,this._expr_44,currVal_44)) {
      this.renderer.setElementProperty(this._el_16,'disabled',currVal_44);
      this._expr_44 = currVal_44;
    }
    const currVal_46:any = import3.inlineInterpolate(1,'',this.parentView.context.title,'');
    if (import3.checkBinding(throwOnChange,this._expr_46,currVal_46)) {
      this.renderer.setText(this._text_19,currVal_46);
      this._expr_46 = currVal_46;
    }
  }
  destroyInternal():void {
    this._vc_35.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import1.AppView<any> {
    if ((nodeIndex == 35)) { return new View_YearPickerComponent2(this.viewUtils,this,35,this._anchor_35,this._vc_35); }
    return (null as any);
  }
  handleEvent_8(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.datePicker.move((0 - 1))) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_16(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.datePicker.toggleMode()) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
  handleEvent_25(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.context.datePicker.move(1)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
var renderType_YearPickerComponent:import2.RenderComponentType = import3.createRenderComponentType('',0,import4.ViewEncapsulation.None,styles_YearPickerComponent,{});
export class View_YearPickerComponent0 extends import1.AppView<import0.YearPickerComponent> {
  _text_0:any;
  _anchor_1:any;
  /*private*/ _vc_1:import10.ViewContainer;
  _TemplateRef_1_5:any;
  _NgIf_1_6:import19.Wrapper_NgIf;
  _text_2:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import1.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_YearPickerComponent0,renderType_YearPickerComponent,import5.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import6.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n',(null as any));
    this._anchor_1 = this.renderer.createTemplateAnchor(parentRenderNode,(null as any));
    this._vc_1 = new import10.ViewContainer(1,(null as any),this,this._anchor_1);
    this._TemplateRef_1_5 = new import17.TemplateRef_(this,1,this._anchor_1);
    this._NgIf_1_6 = new import19.Wrapper_NgIf(this._vc_1.vcRef,this._TemplateRef_1_5);
    this._text_2 = this.renderer.createText(parentRenderNode,'\n  ',(null as any));
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._anchor_1,
      this._text_2
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import17.TemplateRef) && (1 === requestNodeIndex))) { return this._TemplateRef_1_5; }
    if (((token === import20.NgIf) && (1 === requestNodeIndex))) { return this._NgIf_1_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = (this.context.datePicker.datepickerMode === 'year');
    this._NgIf_1_6.check_ngIf(currVal_1_0_0,throwOnChange,false);
    this._NgIf_1_6.ngDoCheck(this,this._anchor_1,throwOnChange);
    this._vc_1.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_1.destroyNestedViews();
  }
  createEmbeddedViewInternal(nodeIndex:number):import1.AppView<any> {
    if ((nodeIndex == 1)) { return new View_YearPickerComponent1(this.viewUtils,this,1,this._anchor_1,this._vc_1); }
    return (null as any);
  }
}
